### PR TITLE
UCP/CONTEXT: If a requested resource is not available, print what is

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11970/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 26-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/82d4e6c1969350e5d1dc19b46140f0c9610c94e5/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/09f1f3a10eae2595e4fe94c7f0d4fa00b840933d/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/5915e4e0bf3ee1379a5d2af8950f77fb6478140d/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/f263e6b5c662572a3c28bc6e96e6d463d7d42a36/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/89f2ab468177dc89012512e1eaf4bc3415270315/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/018125cf2f045ae6ad7bd96b79b7e2bc5117f349/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/3cf252aab3d3a39a8f7156c631293778b1be8e93/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/f0cb5c3bd76ea2d376ba76c80ada552f59a95ead/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/a03bd38770d94482aeaeefdafcf061084abe26d2/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/f40cad24207d0f8df116a5a326bd484aadbd43dc/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/eb89a666753a0f3b0d20e0feea8c8bac4b619001/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/16bc026a3a45578e5afa5d4d333c8b79117c3ef1/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/9870061cfb6aa6caebbf8e9101645789f1d7e240/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/7c2ade5a1131ae3c41bac413ca60cf291b08b59c/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/81042e7ede133f083f52055de5719508cc4752a9/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/9fafbaf91c198f43f4c78fbcca2604b487aef61a/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/8d2bb6a40302c93e950f23c6529fbfe24e5cb2f3/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/08805e042662ae23541daec784653b373e516824/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/77973959fd6051113cbc6e4b1b77697fcf5381cf/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/efdcc1f09e43d0cc27592c40ca6a53203aef2af1/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/6cb206ceff7d1a7395f6d127b714a28b24772804/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/b7af509148996bd280db232798c0abb8dd7bca41/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/d24cc342834ee0774e8b45911b32bba257781d6e/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/ddcb37d92bc2bb4950ebd605eaa2048518c0e215/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/51e1e297c17644f2d61121467739978bd4b23fdb/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/e87c9819531308ffe588560df491e7c8bfb832f3/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/5e3a1ba56ecae01b3574df9bd1274674254d1fd7/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/0d1db348c9fd7ea524895a73cc5625db6f69813e/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/086f119df5ce94555ef054b84e64bb86/raw/50c10d00c8a9851403ac037379ee9037a52407f2/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
@brminich can you pls take a look?

File: src/ucp/core/ucp_context.c [hunk 1]
```diff
@@ -614,16 +623,55 @@ static ucs_status_t ucp_add_tl_resources(ucp_context_h context,
...
+    int found;
+
+    ucs_string_buffer_init(&unavail_strb);
 
+    found = 0;
     for (i = 0; i < cfg->count; i++) {
         if (!(mask & UCS_BIT(i)) && strcmp(cfg->names[i], UCP_RSC_CONFIG_ALL)) {
-            ucs_warn("%s '%s' is not available", title, cfg->names[i]);
+            ucs_string_buffer_appendf(&unavail_strb, "%s'%s'",
+                                       found ? "," : "",
```
Comment (hoopoepg) [inline src/ucp/core/ucp_context.c]: minor: incorrect alignment

File: src/ucp/core/ucp_context.c [hunk 2]
```diff
@@ -1141,33 +1201,43 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
...
+    if (status != UCS_OK) {
+        goto err_free_resources;
     }
 
     ucp_fill_sockaddr_aux_tls_config(context, config);
     ucp_fill_sockaddr_prio_list(context, config);
 
-    uct_release_component_list(uct_components);
-    return UCS_OK;
+    status = UCS_OK;
```
Comment (hoopoepg) [inline src/ucp/core/ucp_context.c]: this line has no effect: status is initialized in line 1221

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11997/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 28-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/4f29763fac4340709dcf37008d15885af9201d66/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/f2fe5f3a90fccc280d5f175d3180cd8172aeb4fa/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/cf99b3ed59a9019e18be207490512ec556fe098d/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/55b443c944a849490754d0bf3969df0d8c5cbc3c/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/00f2333070484df95040b59ac9551a1409fb1a78/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/23cfb3cdafd9e4df2901e32bd3a62682ee8490a9/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/87da62c418909628670393d9adda61ec7afe54c7/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/e7d6ea4425dd040c9db706d9944c683301166d9d/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/2ff89db9f8f53cee5f0ad6fa3ebc8e85d11e227b/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/002ef7cdeda6fd80dc4246b86cb412803989cb6d/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/e1e6ffa8ddda27d397d10dec2a560d245c661151/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/663f2fe4bec9837d87afde938079dcbf7236c4df/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/8e36715e0e154735540d6c645d88c1b4cea41949/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/3a02a5652bcdc9add28a2896b6b0e6da1bead48b/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/061389dca671e19025b721b7c7726c64a938bf11/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/180cab4850cde1e92669df4ac9450e7f7764571f/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/87feff0e7554430354fcd6f15142b1d57d086d0e/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/5ea08f4f53cd5f31ec22b3a6a067effafd52c105/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/7fc4db25689bed399f063c51cafd2e8468d15fec/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/b469295553ee6c3bcc6c4e40ba79803d31bd3f5a/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/fe9f9f5718bcf168c3c48e37064b7d47884fa5b7/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/dc1acd7035fd1957710810b6a505e03cbb7e33d2/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/102c4ae853d92953c55b57c3a5b7bdb068d4c378/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/49e228d90ad800b852b3dac9662ad4c75d787b22/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/0913e3f94f433a93f0ff6d295d5ee4f087422b33/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/8f2936144732731633257db51162dcdbf0ef8d84/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/01cb0f7e5ef82f66df7747d06517ad81befb4ca1/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/9feac4141f61aa8fc27a72cd653e58688059d544/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c31c29625fdab22623017238a36645b8/raw/d72c7ab61e70ecb65d4681f0ac98cbcf945d66e7/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
@hoopoepg ok to merge?

**Pull request merged**